sunpy/map/maphelper.py

```diff
@@ -0,0 +1,246 @@
+                print(k, " : ", acceptable_meta[k])
+            else:
+                print(k, " is not a meta keyword required for sunpy.map.Map \n The map meta keywords are : \n", list(map_meta_keywords.keys()))
+        return
+
+    for k in map_meta_keywords:
+        print(k, " : ", map_meta_keywords[k])
+
+
+@u.quantity_input
+def make_fitswcs_header(data, coordinate, reference_pixel: u.pix = None, scale: u.arcsec/u.pix = None, **kwargs):
+    """
+    Function to create a `sunpy.util.meta.MetaDict` from a coordinate object (`astropy.coordinates.SkyCoord` or coordinate frame)
+    that is required to create a `sunpy.map.Map`
+
+    Parameters
+    ----------
+    data : `~numpy.ndarray`
+        Array data of Map for which a header is required.
+    coordinates : `~astropy.coordinates.SkyCoord`
+        Coordinate object to get meta information for map header.
+    reference_pixel :`~astropy.units.Quantity` of size 2, optional
+        Reference pixel along each axis. These are expected to be Cartestian ordered, i.e
+        the first index is the x axis, second index is the y axis.
+        Defaults to the center of data array, ``(data.shape[0] + 1)/2., (data.shape[1] + 1)/2.)``.
+    scale : `~astropy.units.Quantity` of size 2, optional
+        Pixel scaling along x and y axis (i.e. the spatial scale of the pixels (dx, dy)). These are
+        expected to be Cartestian ordered, i.e [dx, dy].
+        Defaults to ``([1., 1.] arcsec)``.
+    **kwargs:
+        Additional arguments that will be put into the metadict header. These must be in the
+        list of map meta keywords. The keyword arguments for the instrument meta can
+        be given as ``instrument``, ``telescope``, ``observatory``, ``wavelength``, ``exposure``,
+        which will be translated to the accepted meta keywords ``instrume``, ``telescop``,
+        ``obsrvtry``, ``wavelnth``, ``exptime`` respectively.
+
+    Returns
+    -------
+    `~sunpy.util.MetaDict`
+        A `sunpy.util.MetaDict` that contains the header information required for making a `sunpy.map.Map`
+
+    Examples
+    --------
+    >>> from sunpy import map
+    >>> from sunpy.coordinates import frames
+    >>> from astropy.coordinates import SkyCoord
+    >>> from astropy import units as u
+    >>> import numpy as np
+
+    >>> data = np.random.rand(1024, 1024)
+    >>> my_coord = SkyCoord(0*u.arcsec, 0*u.arcsec, obstime="2017-08-01", observer = 'earth', frame=frames.Helioprojective)
+    >>> my_header = map.make_fits_wcs(data, my_coord)
+    >>> my_map = map.Map(data, my_header)
+    """
+
+    if not isinstance(coordinate, SkyCoord):
+        raise ValueError("coordinate needs to be a SkyCoord instance.")
+
+    if coordinate.obstime is None:
+        raise ValueError("The coordinate needs an observation time, `obstime`.")
+
+    coordinate = coordinate.transform_to(frames.Helioprojective)
+
+    meta_wcs = _get_wcs_meta(coordinate)
+
+    meta_observer = _get_observer_meta(coordinate)
```

Comment:
```suggestion
    if hasattr(coordinate, "observer"):
        meta_observer = _get_observer_meta(coordinate)
```

Reply:
oh following on the previous comment - should we just have that if the observer information isn't given we just let map deal with that rather than requiring it here?

Reply:
yes :)